python/python/lance/dataset.py

```diff
@@ -1858,13 +1867,20 @@ class Merge(BaseOperation):
         """
 
         fragments: Iterable[FragmentMetadata]
-        schema: pa.Schema
+        schema: LanceSchema | pa.Schema
```

Comment:
It seems dangerous to still allow this but I think...

This can't lead to invalid behavior now because, if a `pa.Schema` would cause invalid behavior, it will just get caught by the validation checks?

Reply:
Yeah it should get caught now. And it's deprecated. I just didn't want to break the API immediately.